### ISSUE TITLE
Is Conditional Expressions for queries properly setup

### ISSUE THREAD

**Issue by afshawnlotfi** [mhart/dynalite#117]
I tried the following request for querying with a conditional expression:
```
{"ExpressionAttributeNames":{"#ATTR1":"QueryTerm","#ATTR5":"Attendance","#ATTR3":"PostInformation","#ATTR6":"TTL"},"ExpressionAttributeValues":{":val0":{"S":"9q8yy"},":val1":{"N":"1572368826812"}},"FilterExpression":"( ( ( size ( #ATTR5 )) < #ATTR3.Maximum ) AND ( #ATTR6 > :val1 ) )","KeyConditionExpression":"( #ATTR1 = :val0 )","TableName":"ExampleTableName"}```
```
And received the following response from the server:

```
{Amazon.DynamoDBv2.AmazonDynamoDBException: Invalid FilterExpression: Syntax error; Expected ")", "AND", or "OR" but "<" found. ---> Amazon.Runtime.Internal.HttpErrorResponseException
```

The following expression works fine on DynamoDB local offered by Amazon.

**Comment by mhart** [comment]
Hi @afshawnlotfi – does this expression work on DynamoDB in production?

**Comment by afshawnlotfi** [comment]
Haven't tested production but I didn't generate this the [FSharp.AWS.DynamoDB](https://github.com/fsprojects/FSharp.AWS.DynamoDB) library did and I assume they did everything right. The only time I have this exception is when I want to run a Filter expression when updating

**Comment by afshawnlotfi** [comment]
I have noticed that if I do compound conditional expressions for example ```
<@ fun (entry: BroadcastModel) ->
                entry.UserId <> userId && entry.UsersAttending.Count < entry.EventInfo.MaxPeople @>)
```
then this causes a parsing error, however, when I only do a single conditional it works. So maybe there is a bug parsing compound conditionals in ```conditionParser.pegjs```

**Comment by afshawnlotfi** [comment]
I have tested on production now and my query I used above works just fine. There is something wrong with this codebase for Conditional Expressions on Queries and a I assume scans

**Comment by mhart** [comment]
Figured out the issue – it was the extra parens around the `size` function that caused the issue in `( size ( #ATTR5 ))` – I didn't have any tests for that.

Fixed in v3.1.1 – thanks for reporting!

**Comment by afshawnlotfi** [comment]
No problem thanks!
